pkg/tenant/auth.go

```diff
@@ -66,13 +66,53 @@ func (s *Server) Register(c *gin.Context) {
 		AgreePrivacy: params.AgreePrivacy,
 	}
 
+	if params.InviteToken != "" {
+		req.ProjectID = ""
+		req.InviteToken = params.InviteToken
+	}
+
 	var reply *qd.RegisterReply
 	if reply, err = s.quarterdeck.Register(ctx, req); err != nil {
 		sentry.Debug(c).Err(err).Msg("tracing quarterdeck error in tenant")
 		api.ReplyQuarterdeckError(c, err)
 		return
 	}
 
+	// If a member has an invite token, get the member from the database by their email address and update
+	// the member status to Confirmed.
+	if params.InviteToken != "" {
+		var dbMember *db.Member
+		if dbMember, err = db.GetMemberByEmail(c, reply.OrgID, reply.Email); err != nil {
+			sentry.Error(c).Err(err).Msg("could not get member from database by email")
+			c.JSON(http.StatusInternalServerError, api.ErrorResponse("invalid invitation"))
+			return
+		}
+
+		// If the ID from the database does not match the ID from the Register Reply create a new member in the database.
+		if dbMember.ID != reply.ID {
+			if err = db.DeleteMember(c, dbMember.OrgID, dbMember.ID); err != nil {
+				sentry.Error(c).Err(err).Msg("could not delete member from the database")
+				c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not complete user invitation"))
+				return
+			}
+
+			dbMember.ID = reply.ID
+			if err = db.CreateMember(c, dbMember); err != nil {
+				sentry.Error(c).Err(err).Msg("could not recreate member record for invited user")
+				c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not complete user invitation"))
+				return
+			}
+		}
+
+		// Update member status to Confirmed.
+		dbMember.Status = db.MemberStatusConfirmed
```

Comment:
I think we also need to update this record with the name from the request, since we didn't have it when the invite was created. Also we should set `DateAdded` here to the current time.